django/librarian/models.py

```diff
@@ -377,15 +382,16 @@ def delete(self, *args, **kwargs):
             logger.error(f"Failed to remove document from vector store: {e}")
         super().delete(*args, **kwargs)
 
-    def process(self):
+    def process(self, user: User = None):
         from .tasks import process_document
 
         # Logic for updating the document embeddings, metadata, etc.
         if not (self.file or self.url):
             self.status = "ERROR"
             self.save()
             return
-        process_document.delay(self.id, get_language())
+        # User is optionally passed through for cost tracking purposes
+        process_document.delay(self.id, user.upn if user else None, get_language())
```

Comment:
There won't always be a user to attribute the cost to. e.g. when loading documents from a setup script, or scheduled refresh.